src/asynqp/channel.py

```diff
@@ -304,11 +311,14 @@ def receive_deliver(self, frame):
             payload.routing_key,
             payload.consumer_tag
         )
-        self.reader.ready()
 
-        yield from self.async_receive()
+        # Deliver the message to consumers when done
+        def callback(tag_msg):
+            self.consumers.deliver(*tag_msg)
+            self.reader.ready()
```

Comment:
These callbacks don't seem to access the closure so there's no need for them to be nested. I'd prefer to see it as proper (private) method on the `MessageReceiver`.

I understand that it dispatches to a different function depending on whether it's a basic get or a basic deliver ([line 300](https://github.com/benjamin-hodgson/asynqp/pull/50/files#diff-a12f879848a4250e1a7b48cb26ff7fe5R300)). I find that slightly confusing. I'd prefer to have a Boolean property called something like `is_deliver`.